internal/cinema/cinemaserver.go

```diff
@@ -63,8 +72,13 @@ func (s *Server) Routes(ctx context.Context) *mux.Router {
 	r := mux.NewRouter()
 
 	// Middleware
+	// TODO: request id?
+	r.Use(middleware.RecoverPanic())
 	r.Use(middleware.AttachAppLogger(logger))
 	r.Use(middleware.ProcessMaintenance(s.cfg))
+	r.Use(middleware.ApplyAuthorization(s.cfg))
+	r.Use(middleware.ApplyAuthentication())
```

Comment:
Check: Why is authentication after authorization. Does the order matter?

Reply:
Yes, order matters. They are executed in that order. Authentication requires authorization to provide the token.

Reply:
But doesn't that mean that you could get a 403 when you should get a 401?

Reply:
Kind of what authentication means, but as I see it authentication is that you have a JWT, and authorization is that the JWT contains the correct claims, such as scopes. 

1. Validera att anropet är korrekt
2. Validera att access token är korrekt
3. Transformera access token till rättighetsmodell
4. Validera att data i anropet är korrekt
5. Validera behörighet att utföra operationen
6. Validera behörighet till det data som efterfrågas eller påverkas

Step 2 in this list is: 
* Korrekt kryptografisk signatur
* Signerad av rätt IdP (oftast konfigurerad som en URL till vår IdP)
* Utställd med en audience som gäller för vårt API
* Korrekt typ, dvs. en access token och inte någonting annat
* Giltig, t ex med avseende på tid

But authorization for me is when you have for example "movies.create" scope.

Reply:
But I trust that you know that it's in the correct order, I have no clue :grin:

Reply:
hm I don't think we're actually verifying that we get an access token and not an id_token,

but 1 and 2 is the authorization middleware in this PR, 3 is basically what the authentication middleware is going to do (transform the access token to an id_token).

I will look into checking the type though...